packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -154,36 +154,50 @@ class CallWithChatContext {
  */
 export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapter {
   private callAdapter: CallAdapter;
-  private chatAdapterPromise: Promise<ChatAdapter>;
   private chatAdapter: ChatAdapter | undefined;
   private context: CallWithChatContext;
+  private emitter: EventEmitter = new EventEmitter();
   private onChatStateChange: (newChatAdapterState: ChatAdapterState) => void;
   private onCallStateChange: (newChatAdapterState: CallAdapterState) => void;
 
-  constructor(callAdapter: CallAdapter, chatAdapterPromise: Promise<ChatAdapter>) {
+  constructor(callAdapter: CallAdapter, chatAdapter?: ChatAdapter) {
     this.bindPublicMethods();
     this.callAdapter = callAdapter;
     this.context = new CallWithChatContext(callWithChatAdapterStateFromBackingStates(callAdapter));
 
     const onChatStateChange = (newChatAdapterState: ChatAdapterState): void => {
       this.context.updateClientStateWithChatState(newChatAdapterState);
     };
-    this.onChatStateChange = onChatStateChange;
 
-    this.chatAdapterPromise = chatAdapterPromise;
-    this.chatAdapterPromise.then((chatAdapter) => {
-      chatAdapter.onStateChange(this.onChatStateChange);
+    this.onChatStateChange = onChatStateChange;
+    if (chatAdapter) {
       this.chatAdapter = chatAdapter;
```

Comment:
this is the same code as when the promise resolves - move code block to a common private "chat adapter available" function so one isn't updated without the other